web/ui/react-app/src/api/Fetch.tsx

```diff
@@ -0,0 +1,75 @@
+import { Component } from 'react';
+import { fetchAPI } from './FetchAPI';
+
+export interface FetchState<T> {
+  data?: T;
+  error?: Error;
+}
+
+interface FetchProps {
+  url?: string;
+  urls?: string[];
+  options?: RequestInit;
+  children: (state: any) => any;
```

Comment:
Any way to make `any` more strongly typed? I'm thinking about how I might use this as a client. `any` tends to obscure usage for me.

Reply:
I think no..any data can be passed to the consumer. I mean I don't know whether the type can be predicted. Any ideas?

Reply:
You could do something like 
```
interface FetchProps<T, S> {
   ...
   children: (state: S) => T;
}

or something like that

Reply:
How can I use this here ```...extends Component<FetchProps<???>>?

Reply:
Yeah I'm not totally sure either. No worries if it's too much of a pain. Would be cool if you can figure it out

Reply:
I'll try, thanks!